sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -529,6 +529,9 @@ static void _handle_rtr_sol(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
         _snd_rtr_advs(netif, &ipv6->src, false);
     }
 #endif  /* GNRC_IPV6_NIB_CONF_6LR */
+    else {
+        (void)nce;  /* NCE is not used */
+    }
```

Comment:
Put that inside the `#if GNRC_IPV6_NIB_CONF_6LR` block, you don't need the `else` then.
It's only unused because in this case `gnrc_netif_is_6ln(netif)` is defined to `false` - it should still generate a warning if it becomes unused otherwise.

Reply:
Mh, but `gnrc_netif_is_6ln(netif)` is only not always false `#if GNRC_IPV6_NIB_CONF_6LN`, so there are cases, when `#if !GNRC_IPV6_NIB_CONF_6LR` and `nce` is not used.

Reply:
(Mind `R`[outer] vs `N`[ode])

Reply:
Oh, tricky!
I'd still think 

```C
#if GNRC_IPV6_NIB_CONF_6LN
    (void) nce;
#endif
```

Would be nicer than the strange nopy-else, so a proper warning can still be triggered if the variable becomes unused in an unexpected way.

Reply:
Ok

Reply:
No need for the `else` though - it will count as used no matter if it's in a (reachable) branch or not.

Reply:
True dat.